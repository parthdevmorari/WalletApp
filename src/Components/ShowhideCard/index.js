
import component from './showHideCard';
export default component;