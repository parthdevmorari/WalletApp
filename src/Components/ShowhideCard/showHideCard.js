
import { View, Text, Image,TouchableOpacity } from 'react-native'
import React from 'react'
import styles from './showHideCardStyle'

const ShowhideNumber = (props) => {
    const { title, imageSource ,style,onPress} = props;
    return (
        <TouchableOpacity style={[styles.showContainer,style]}  onPress={()=>{onPress()}} >
            <View style = {styles.imageContainer}>
                <Image source={imageSource} style={styles.imgStyle} />
            </View>

            <Text style={styles.txt}>{title}</Text>


        </TouchableOpacity>
    )
}

export default ShowhideNumber