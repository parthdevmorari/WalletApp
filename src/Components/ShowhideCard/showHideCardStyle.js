

import { StyleSheet, StatusBar } from 'react-native';
import { widthPercentageToDP as wp, heightPercentageToDP as hp } from 'react-native-responsive-screen';
import { color } from '../../Constants/Color'
import { theme } from '../../Constants/Theme'
const styles = StyleSheet.create({
    
	showContainer: {
        backgroundColor: color._WHITE,
        paddingHorizontal: wp(5),
        paddingBottom: wp(5.5),
        borderTopLeftRadius: wp(1.2),
        borderTopRightRadius: wp(1.2),
        alignSelf: "flex-end",
        flexDirection:'row',
        alignItems:'center'

    },
    txt: {
        color: color._01D167,
        textAlign: "center",
        paddingLeft: wp(1.25),
        fontSize: wp(3),
        fontFamily: "AvenirNext",

    },
    imageContainer: {
        width : wp(5),
        height: wp(5),

    },
    imgStyle: {
        width : "100%",
        height: "100%",
    }


});
export default styles;