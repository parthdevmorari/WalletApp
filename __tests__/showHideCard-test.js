import 'react-native';
import React from 'react';
import ShowhideCard from '../src/Components/ShowhideCard/showHideCard'
import Enzyme, {shallow, mount} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from 'enzyme-to-json';

Enzyme.configure({adapter: new Adapter()});

describe('ShowhideCard Component', () => {
  test('ShowhideCard Component', () => {
    const wrapper = shallow(<ShowhideCard />);
    expect(wrapper.exists()).toBe(true);
  });
  it('ShowhideCard  toMatchSnapshot', () => {
    const wrapper = shallow(<ShowhideCard />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});