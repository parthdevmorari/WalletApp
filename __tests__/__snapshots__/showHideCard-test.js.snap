// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowhideCard Component ShowhideCard  toMatchSnapshot 1`] = `
<TouchableOpacity
  onPress={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "flex-end",
        "backgroundColor": "#FFFFFF",
        "borderTopLeftRadius": 9,
        "borderTopRightRadius": 9,
        "flexDirection": "row",
        "justifyContent": "center",
        "minWidth": 300,
        "paddingBottom": 41.5,
        "paddingTop": 11.5,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "height": 37.5,
        "width": 37.5,
      }
    }
  >
    <Image
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "color": "#01D167",
        "fontSize": 22.5,
        "fontWeight": "700",
        "paddingLeft": 9.5,
        "textAlign": "center",
      }
    }
  />
</TouchableOpacity>
`;
